packages/core/src/components/graphs/wordcloud.ts

```diff
@@ -106,18 +118,20 @@ export class WordCloud extends Component {
 					return self.model.getFillColor(d[groupMapsTo], d.text, d);
 				})
 				.attr('text-anchor', 'middle')
-				.transition(
-					self.services.transitions.getTransition(
-						'wordcloud-text-update-enter',
-						animate
-					)
+				.transition()
+				.call((t) =>
+					self.services.transitions.setupTransition({
+						transition: t,
+						name: 'wordcloud-text-update-enter',
+						animate,
+					})
 				)
 				.attr('transform', (d) => `translate(${d.x}, ${d.y})`)
-				.attr('opacity', 1);
+				.attr('opacity', 1)
+				.on('end', () => {
+					self.componentContainer.classed('updating', false);
+				});
```

Comment:
Class updating is being removed (By setting to false) from the element n number of times (In demo example, 42).

Switch approach to async/await or promises.

```suggestion
				.end();
				
				//Assign merged text to promise
				promise.then(() => {
				  // Give users the option to hover and select legend after everything loads
				  self.addEventListeners(mergedText);
				  self.componentContainer.classed('updating', false); 
				});
```